Kernel/FileSystem/FATFS/FileSystem.h

```diff
@@ -17,6 +17,42 @@
 
 namespace Kernel {
 
+class DOSBIOSParameterBlock final {
+private:
+    KBuffer const* const m_boot_record;
+    const struct DOS4BIOSParameterBlock* dos4_block;
+    const struct DOS7BIOSParameterBlock* dos7_block;
+
+public:
+    DOSBIOSParameterBlock(KBuffer const* const boot_record)
+        : m_boot_record(boot_record)
+        , dos4_block(reinterpret_cast<const struct DOS4BIOSParameterBlock*>(boot_record->bytes().offset_pointer(0x024)))
+        , dos7_block(reinterpret_cast<const struct DOS7BIOSParameterBlock*>(boot_record->bytes().offset_pointer(0x024)))
+        , block(reinterpret_cast<const struct DOS3BIOSParameterBlock*>(boot_record->bytes().data()))
+    {
+    }
+    const struct DOS3BIOSParameterBlock* block;
+
+    DOSBIOSParameterBlockVersion get_bpb_version() const;
+
+    DOS4BIOSParameterBlock const* get_dos4_bpb() const;
+
+    DOS7BIOSParameterBlock const* get_dos7_bpb() const;
```

Comment:
to better match the style of the project, these should be updated as such:

- block should be a private member
- all private members of the class should be prefixed with `m_`
- the getters can omit the `get_` i.e. `DOS3BIOSParameterBlock const* block() const` and `DOS4BIOSParameterBlock const* dos4_bpb() const`.

I also think it might make sense to only have the boot record as a member, and just return the calculated offset pointer in the getters, but that's not as important.

Reply:
Cleaned up and squashed into f043062a9dd27c3c7ea0fb31effa026b27cab096.